scipy/stats/tests/test_stats.py

```diff
@@ -4808,118 +4808,125 @@ def test_location_sign(self, ksfunc, alternative,
-    rvs2_2D = np.array([np.linspace(1.01,99.989,100), np.linspace(1,100,100)])
-
-    t,p = stats.ttest_rel(rvs1, rvs2, axis=0)
-    assert_array_almost_equal([t,p],(tr,pr))
-    t,p = stats.ttest_rel(rvs1_2D.T, rvs2_2D.T, axis=0)
-    assert_array_almost_equal([t,p],tpr)
-    t,p = stats.ttest_rel(rvs1_2D, rvs2_2D, axis=1)
-    assert_array_almost_equal([t,p],tpr)
-
-    # test scalars
-    with suppress_warnings() as sup, \
-            np.errstate(invalid="ignore", divide="ignore"):
-        sup.filter(RuntimeWarning, "Degrees of freedom <= 0 for slice")
-        t, p = stats.ttest_rel(4., 3.)
-    assert_(np.isnan(t))
-    assert_(np.isnan(p))
+@pytest.mark.skip_xp_backends(cpu_only=True,
+                              reasons=['Uses NumPy for pvalue, CI'])
+@pytest.mark.usefixtures("skip_xp_backends")
+@array_api_compatible
+class TestTTestRel:
+    def test_regression(self, xp):
+        # regression test
+        tr, pr = 0.81248591389165692, 0.41846234511362157
+        tpr = xp.asarray([[tr, -tr], [pr, pr]])
+
+        rvs1 = xp.linspace(1, 100, 100)
+        rvs2 = xp.linspace(1.01, 99.989, 100)
+        rvs1_2D = xp.stack([xp.linspace(1, 100, 100), xp.linspace(1.01, 99.989, 100)])
+        rvs2_2D = xp.stack([xp.linspace(1.01, 99.989, 100), xp.linspace(1, 100, 100)])
+
+        t, p = stats.ttest_rel(rvs1, rvs2, axis=0)
+        xp_assert_close(xp.asarray([t, p]), xp.asarray([tr, pr]))
+        t, p = stats.ttest_rel(rvs1_2D.T, rvs2_2D.T, axis=0)
+        xp_assert_close(xp.asarray([t, p]), tpr)
+        t, p = stats.ttest_rel(rvs1_2D, rvs2_2D, axis=1)
+        xp_assert_close(xp.asarray([t, p]), tpr)
+
+        # test scalars
+        with suppress_warnings() as sup, \
+                np.errstate(invalid="ignore", divide="ignore"):
+            sup.filter(RuntimeWarning, "Degrees of freedom <= 0 for slice")
+            t, p = stats.ttest_rel(4., 3.)
+        assert xp.isnan(t)
+        assert xp.isnan(p)
 
-    # test for namedtuple attribute results
-    attributes = ('statistic', 'pvalue')
-    res = stats.ttest_rel(rvs1, rvs2, axis=0)
-    check_named_results(res, attributes)
+        # test for namedtuple attribute results
+        attributes = ('statistic', 'pvalue')
+        res = stats.ttest_rel(rvs1, rvs2, axis=0)
+        check_named_results(res, attributes)
 
-    # test on 3 dimensions
-    rvs1_3D = np.dstack([rvs1_2D,rvs1_2D,rvs1_2D])
-    rvs2_3D = np.dstack([rvs2_2D,rvs2_2D,rvs2_2D])
-    t,p = stats.ttest_rel(rvs1_3D, rvs2_3D, axis=1)
-    assert_array_almost_equal(np.abs(t), tr)
-    assert_array_almost_equal(np.abs(p), pr)
-    assert_equal(t.shape, (2, 3))
+        # test on 3 dimensions
+        rvs1_3D = xp.dstack([rvs1_2D, rvs1_2D, rvs1_2D])
+        rvs2_3D = xp.dstack([rvs2_2D, rvs2_2D, rvs2_2D])
+        t, p = stats.ttest_rel(rvs1_3D, rvs2_3D, axis=1)
+        xp_assert_close(xp.abs(t), xp.full(t.shape, tr))
+        xp_assert_close(xp.abs(p), xp.full(p.shape, pr))
```

Comment:
```suggestion
        xp_assert_close(p, xp.full(p.shape, pr))
```
I hope we're not returning negative p-values!

Reply:
I thought I should leave htis open because it applies elsewhere.